### ISSUE TITLE
[Bug] "ITMS-90809: Deprecated API Usage" in Xamarin.Youtube.Player.iOS

### ISSUE THREAD

**Issue by HealsCodes** [xamarin/XamarinComponents#812]
After adding the NuGet package for _Xamarin.YouTube.Player.iOS_ to my project and submitting a version to TestFlight I receive the following mail from Apple:

>Dear Developer,
>
>We identified one or more issues with a recent delivery for your app, "xxxxxxx" x.x.x (x.x.x.x). Your delivery was successful, but you may wish to correct the following issues in your next delivery:
> 
> **ITMS-90809: Deprecated API Usage** - Apple will stop accepting submissions of apps that use UIWebView APIs starting from December 2020 . See https://developer.apple.com/documentation/uikit/uiwebview for more information.
> 
> After you’ve corrected the issues, you can upload a new binary to App Store Connect.
>
> Best regards,

Taking a quick look over the sources it seems that the YouTube component for iOS is still using / referring to the now long deprecated `UIWebView` instead of using `WKWebView`.

**NuGet version:** _Xamarin.YouTube.Player.iOS v0.1.6_

**Comment by gadeweever** [comment]
This issue also seems present in Xamarin.iOS.MaterialComponents.

```
grep -r "UIWebView" .

Binary file ./iOS/obj/iPhoneSimulator/Debug/mtouch-cache/MaterialComponents.framework/MaterialComponents matches
Binary file ./iOS/bin/iPhone/Release/.../Frameworks/MaterialComponents.framework/MaterialComponents matches
Binary file ./iOS/bin/iPhoneSimulator/Debug/.../Frameworks/MaterialComponents.framework/MaterialComponents matches
```

**Comment by HealsCodes** [comment]
@gadeweever which version does this relate to? I'm also using MaterialComponents (72.2.0.1) and don't get a reference:

```bash
./packages/Xamarin.iOS.MaterialComponents.72.2.0.1/lib/Xamarin.iOS$ grep -i UIWebView MaterialComponents.dll
./packages/Xamarin.iOS.MaterialComponents.72.2.0.1/lib/Xamarin.iOS$ strings MaterialComponents.dll | grep -c -i UIWebView 
0
```

Compared to YouTube.Player.iOS (0.1.6):
```bash
./packages/Xamarin.YouTube.Player.iOS.0.1.6/lib/XamariniOS10$ grep UIWebView YouTube.Player.iOS.dll
Binary file YouTube.Player.iOS.dll matches
./packages/Xamarin.YouTube.Player.iOS.0.1.6/lib/XamariniOS10$ strings YouTube.Player.iOS.dll | grep -c UIWebView
117
```

**Comment by gadeweever** [comment]
@Shirk This is on 72.2.0.1. Does the grep change if you also run it on your build output folder?

EDIT: According to this: https://github.com/xamarin/XamarinComponents/issues/745 the pre-release package for MaterialComponents should not have this reference. Consider my addition off-topic.

**Comment by Rishi2611** [comment]
@shirk 
Any solution for this  I am facing same issue

**Comment by HealsCodes** [comment]
> Any solution for this I am facing same issue

No, none - ended up writing a wrapper around YouTube's iframe API that exposes the javascript interface as a native class.
